src/main/java/uk/gov/hmcts/divorce/solicitor/event/Applicant2SolicitorRespondRequestForInformation.java

```diff
@@ -0,0 +1,121 @@
+                .name("Submit Response")
+                .description("Submit response")
+                .showSummary()
+                .showEventNotes()
+                .endButtonLabel("Submit")
+                .aboutToSubmitCallback(this::aboutToSubmit)
+                .grant(CREATE_READ_UPDATE, APPLICANT_2_SOLICITOR)
+                .grantHistoryOnly(CASE_WORKER, SUPER_USER, LEGAL_ADVISOR, JUDGE))
+                .page("requestForInformationResponse", this::midEvent)
+                .pageLabel("Submit Response")
+                .complex(CaseData::getRequestForInformationList)
+                    .complex(RequestForInformationList::getRequestForInformationResponseApplicant2Solicitor)
+                        .optional(RequestForInformationResponseDraft::getRfiDraftResponseDetails)
+                        .optional(RequestForInformationResponseDraft::getRfiDraftResponseDocs)
+                    .done()
+                .done();
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> midEvent(CaseDetails<CaseData, State> details,
+                                                                  CaseDetails<CaseData, State> detailsBefore) {
+
+        log.info("{} midEvent callback invoked for Case Id: {}", APP_2_SOLICITOR_RESPOND_REQUEST_INFO, details.getId());
+
+        RequestForInformationResponseDraft response =
+            details.getData().getRequestForInformationList().getRequestForInformationResponseApplicant2Solicitor();
+        List<ListValue<DivorceDocument>> responseDocs = response.getRfiDraftResponseDocs();
+        String responseDetails = response.getRfiDraftResponseDetails();
+
+        if ((responseDocs == null || responseDocs.isEmpty()) && (responseDetails == null || responseDetails.isEmpty())) {
+            return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+                .errors(Collections.singletonList(MUST_ADD_DOCS_OR_DETAILS_ERROR))
+                .build();
+        }
+
+        return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+            .data(details.getData())
+            .build();
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<CaseData, State> details,
+                                                                       CaseDetails<CaseData, State> beforeDetails) {
+        log.info("{} about to submit callback invoked for Case Id: {}", APP_2_SOLICITOR_RESPOND_REQUEST_INFO, details.getId());
+
+        CaseData data = details.getData();
+        RequestForInformationResponse requestForInformationResponse = new RequestForInformationResponse();
+
+        if (isApplicant2Solicitor(details.getId())) {
+            requestForInformationResponse.setValues(data, APPLICANT2SOLICITOR);
+        } else {
+            return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+                .errors(Collections.singletonList(UNABLE_TO_SUBMIT_RESPONSE_ERROR + details.getId()))
+                .build();
+        }
```

Comment:
I'd be tempted to remove this check as they have to be applicant 2s solicitor to use the event and ccdAccessService seems to send requests to get the roles, making it relatively expensive